### DISCUSSION TITLE
how should i get all properties and functions of object in Jint like js in browser console

### DISCUSSION THREAD

**Discussion by Zaitonn** [sebastienros/jint#1415] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/103164490/214294840-65990873-a61a-4691-b0cf-aecf2b5eafd7.png)
![image](https://user-images.githubusercontent.com/103164490/214294980-1d40a525-2e02-422a-9125-ebe72bf77a65.png)

has there been a method in Jint to get them like these?
if not, how can i get all properties of object in Jint

```cs
engine.SetValue("serein_log",
  new Action<JsValue>((content) => Logger.Output(LogType.Plugin_Info, $"[{@namespace}]", content)));
```
what `object.ToString()` in NET gave me is thie class name like `System.Dynamic.ExpandoObject`.  i got `[object Object]` through `JsValue.ToString()`  and `JSON.stringify` in js did't gave me functions :(

**Comment by lahma** [comment, marked as answer]
Probably easiest to serialize as JSON like this:

```c#
JsValue result = new Jint.Native.Json.JsonSerializer(engine).Serialize(theObjectInQuestion);
// can call ToString() for result
```

**Reply by Zaitonn** [reply]
thanks
btw is it possible to get all function names of an object?

**Reply by lahma** [reply]
Basic JS solutions like this should work: https://stackoverflow.com/a/35510680/111604

**Reply by Zaitonn** [reply]
thanks a lot
